pylint/checkers/base.py

```diff
@@ -1810,9 +1816,35 @@ def _check_logical_tautology(self, node):
             suggestion = "%s %s %s" % (left_operand, operator, right_operand)
             self.add_message('comparison-with-itself', node=node, args=(suggestion,))
 
+    def _check_callable_comparison(self, node):
+        operator = node.ops[0][0]
+        if operator not in ('==', '!=', '<', '>', '<=', '>='):
+            return
+
+        def _is_bare_callable(node):
+            bare_callables = (astroid.FunctionDef, astroid.BoundMethod)
+            if (isinstance(node, bare_callables) or
+                    isinstance(utils.safe_infer(node), bare_callables)):
+                return True
+            return False
+
+        left_operand, right_operand = node.left, node.ops[0][1]
+        bare_callables_count = [True for operand in (left_operand, right_operand)
```

Comment:
Instead of holding a list, you can do something like `sum(True for operand in ...)` and then verify that the value is 1.